src/reflection.ts

```diff
@@ -688,4 +733,168 @@ export class Parser {
       return result;
     };
   }
+
+  readVectorOfUnionLambda(
+    field: reflection.Field,
+    discriminatorField: reflection.Field,
+    readDefaults: boolean,
+  ): (table: Table) => Record<string, any>[] {
+    const fieldType = field.type();
+    if (fieldType === null) {
+      throw new Error(`Malformed schema: "type" field of '${field.name()}' not populated.`);
+    }
+
+    // This will deserialize the vector of discriminator values
+    const readDiscriminators = this.readVectorOfScalarsLambda(discriminatorField);
+
+    // This will read the vector of tables from the actual field
+    // How to deserialize each table will depend on the discriminator value at the same vector
+    // index position in the discriminator values field
+    const vectorLambda = this.readVectorOfTablesLambda(field);
+
+    const unionDeserializers = this.createUnionDeserializers(fieldType, readDefaults);
+
+    return (table: Table) => {
+      const res = readDiscriminators(table);
+      if (!res) {
+        throw new Error(
+          `Malformed message: missing vector discriminators field: ${discriminatorField.name()}`,
+        );
+      }
+
+      const vector = vectorLambda(table);
+      if (vector == null) {
+        throw new Error(`Malformed message: missing vector table for field: ${field.name()}`);
+      }
+
+      if (res.length !== vector.length) {
+        throw new Error(
+          `malformed message: ${field.name()} length != ${discriminatorField.name()} length`,
+        );
+      }
+
+      const result = [];
+      for (let idx = 0; idx < vector.length; ++idx) {
+        const discriminator = res[idx];
+        if (typeof discriminator !== "number") {
+          throw new Error(
+            `Malformed union discriminator value is not a number in field: ${discriminatorField.name()}`,
+          );
+        }
+
+        // Skip NONE
+        if (discriminator < 0) {
```

Comment:
what would the entry in the vector of tables be in this case and did we handle it properly?

Reply:
The types generated for the tests did not allow pushing undefined to the vector so I suspect it might not be a thing.